cdb2api/cdb2api.c

```diff
@@ -5081,9 +5115,53 @@ int cdb2_clearbindings(cdb2_hndl_tp *hndl)
     free(hndl->bindvars);
     hndl->bindvars = NULL;
     hndl->n_bindvars = 0;
+    free(hndl->bind_list_name);
+    hndl->bind_list_name = NULL;
+    free(hndl->bind_list_repl);
+    hndl->bind_list_repl = NULL;
     return 0;
 }
 
+/* bind @list with array of type/variable */
+int cdb2_bind_list(cdb2_hndl_tp *hndl, const char *varname, int count,
+                   int type[], const void *varaddr[], int length[])
+{
+    free(hndl->bind_list_name);
+    free(hndl->bind_list_repl);
+    int len = strlen(varname);
+    hndl->bind_list_name = malloc(len + 1);
+    hndl->bind_list_name[0] = '@';
+    strcpy(&hndl->bind_list_name[1], varname); // to search
```

Comment:
Unless I'm misreading, there's an off by one here and this needs to `malloc(len + 2)`.  I believe this overflows the buffer - given `varname="x"`, I think this allocates 2 bytes but writes `{'@', 'x', '\0'}`